samase.py

```diff
@@ -1,34 +1,19 @@
 import os
 import subprocess
-import enum
 
-SAMASE = "bin\\samase.exe"
-SAMASE64 = "bin\\samase64.exe"
+def internal_run(args: list, is64: bool):
+    SAMASE = "bin\\samase.exe"
+    SAMASE64 = "bin\\samase64.exe"
 
-class OpType(enum.Enum):
-    Run = enum.auto(), # ex: samase.exe C:\samase\uedaip
-    Build = enum.auto(), # ex: samase.exe C:\samase\uedaip --pack C:\samase\UEDAIP_BroodWar.exe --pack_version "23.2212" --pack_icon uedr.ico
-    Unpack = enum.auto(), # ex: samase.exe C:\samase\unpack --unpack C:\samase\unpack.exe
+    path = [SAMASE if not is64 else SAMASE64, *args]
+    print(f"Samase string debug: {path}")
+    subprocess.run([path, *args])
 
-def samase_line_constructor(operation_type: OpType, path:str, is64: bool = False):
-    local_samase = SAMASE if not is64 else SAMASE64
-    match operation_type:
-        case OpType.Run:
-            return [local_samase, path]
-        case OpType.Build:
-            return [local_samase, path, "--pack", os.getcwd()+"\\packed.exe"]
-        case OpType.Unpack:
-            os.chdir("target")
-            t_dir = os.getcwd()
-            os.chdir("..")
-            return [local_samase, t_dir, "--unpack", path]
+def run(path: str, is64: bool):
+    internal_run([path], is64)
```

Comment:
With recognizing the `operation_type` as basically switching over separate functions, we can literally have them as 3 separate functions, and having them call one internal handler function which does logging plus chooses the right exe, we end up with a simpler API for the samase module.